src/util/shm/mpir_shm_coll.c

```diff
@@ -3,36 +3,156 @@
  *  (C) 2019 by Argonne National Laboratory.
  *      See COPYRIGHT in top-level directory.
  */
+#include <mpidimpl.h>
 #include "mpir_shm.h"
 #include "mpir_shm_coll.h"
+#include "mpir_mem.h"
+#include "build_nodemap.h"
+
+#define PROC_MEM_LEN (MPIR_SHM_CACHE_LINE_LEN * 16)
```

Comment:
@raffenet in the case of `bc` exchange I think this is not even close to be sufficient as `VALLEN` is `1024` and every process needs `VALLEN * size / local_size`.